src/main/java/com/teragrep/jos_01/procfs/Process.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import com.teragrep.jos_01.procfs.status.ProcessStat;
+import com.teragrep.jos_01.procfs.status.Statm;
+
+public class Process {
+
+    private final long processId;
+    private final File procDirectory;
+
+    public Process(String processId) {
+        this(Integer.parseInt(processId));
+    }
+
+    public Process(long processId) {
+        this(processId, "/proc");
+    }
+
+    public Process(long processId, String procDirectoryPath) {
+        this.processId = processId;
+        this.procDirectory = new File(procDirectoryPath, Long.toString(processId));
+    }
+
+    // Creates a Status object based on the chosen file name in /proc for this process. Overloaded to accept different kinds of Status objects
+    private ArrayList<String> reportStatistics(String procFileName) {
+        ArrayList<String> rows = readProcFile(procFileName);
+        return rows;
+    }
+
+    private ArrayList<String> readProcFile(String procFileName) {
+        RowFile procFile = new RowFile(procDirectory, procFileName);
+        ArrayList<String> rows = procFile.readFile();
+        return rows;
+    }
+
+    public ProcessStat stat() {
+        ArrayList<String> rows = readProcFile("stat");
+        return new ProcessStat(rows);
+    }
+
+    public Statm statm() {
+        ArrayList<String> rows = readProcFile("statm");
+        return new Statm(rows);
+    }
+
+    public ArrayList<String> availableProcFiles() {
+        ArrayList<String> fileNames = procFileNames(new ArrayList<String>(), procDirectory);
+        return fileNames;
+    }
+
+    private ArrayList<String> procFileNames(ArrayList<String> nameList, File file) {
+        if (!file.isDirectory()) {
+            nameList.add(file.getPath().replace(procDirectory.getPath(), ""));
+        }
+        else {
+            File[] subdirectories = file.listFiles();
+            if (subdirectories == null) {
+                System.out
+                        .println(
+                                "Failed to get all file names! Either no permission to open file at " + file.getPath()
+                                        + " or it is not a directory!"
+                        );
+                return nameList;
+            }
+            for (File child : file.listFiles()) {
+                procFileNames(nameList, child);
```

Comment:
Does the it modify the current list or is this a noop?

Reply:
This will modify the list, recursively adding filenames of all child files to the list, finally returning a list containing all filenames to which we have permission to access.